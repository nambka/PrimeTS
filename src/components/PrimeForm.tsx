import React, { useState } from 'react';
import { calcPrime } from './Algo';

const PrimeForm = () => {
  const [inputNum, setInputNum] = useState('');
  const [answer, setAnswer] = useState('');

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputNum(e.target.value);
  }

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const num: number = +inputNum
    setAnswer(calcPrime(num));
  }

  return (
    <form id="frmFindHighestPrime" onSubmit={handleSubmit} className="container-fluid nambk-container-input text-center">
      <h2 className="nambk-header">Highest Prime</h2>
      <div className="nambk-header-desc">Lower than Input Number</div>
      <div>
        <input id="inputNumber" name="inputNumber" className="nambk-input" style={{width: '50%'}} 
                type="number" value={inputNum} aria-label="Input Number" autoFocus
                onChange={handleChange} placeholder="Enter a number"/>
        <input id="btnFindHighestPrime" className="nambk-btn nambk-btn-primary" 
                type="submit" value="Go"/>
      </div>
      <div id="result" className="mt-3" aria-readonly style={{minHeight: '2rem',}}>
        {answer}  
      </div>   
    </form>
  );
}
export default PrimeForm;