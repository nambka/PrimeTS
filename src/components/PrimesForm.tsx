import React from 'react';
import { sieve } from './Algo';

// Render the input form to enter two numbers
interface IProps {
  // value: number;
}
interface IState {
  inputFirstNum: string;
  inputLastNum: string;
  answer: (string | number)[];
}

class PrimesForm extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      inputFirstNum: '1',
      inputLastNum: '1000',
      answer: ['Primes']
    };
    // this.onFirstNumChange = this.onFirstNumChange.bind(this);
    // this.onLastNumChange = this.onLastNumChange.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  // onFirstNumChange(e: React.ChangeEvent<HTMLInputElement>) {
  //   this.setState({
  //     inputFirstNum: +e.target.value
  //   })
  // }

  // onLastNumChange(e: React.ChangeEvent<HTMLInputElement>) {
  //   this.setState({
  //     inputLastNum: +e.target.value
  //   })
  // }

  handleChange(e: any) {
    this.setState({
      [e.target.name]: e.target.value,
    } as IState);
  }

  handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
    const num1 = Number(this.state.inputFirstNum)
    const num2 = Number(this.state.inputLastNum)
    const max = num1 > num2 ? num1 : num2
    const min = max === num1 ? num2 : num1

    this.setState({
      answer: sieve(max, min),
    });   
  }

  render() {
    return (
      <form id="frmFindPrimes" onSubmit={this.handleSubmit} className="container-fluid nambk-container-input">
        <h2 className="nambk-header">Prime Numbers</h2>
        <div className="nambk-header-desc">Between Two Numbers</div>
        <div>
          <input className="nambk-input" style={{width: '33%'}} 
                 type="number" min="1" max="9999999" aria-label="Input Number"
                 value={this.state.inputFirstNum} onChange={this.handleChange.bind(this)} 
                 id="inputFirstNum" name="inputFirstNum" placeholder="Enter a number"/>
          <input className="nambk-input" style={{width: '33%'}} 
                 type="number" min="1" max="9999999" aria-label="Input Number"
                 value={this.state.inputLastNum} onChange={this.handleChange.bind(this)} 
                 id="inputLastNum" name="inputLastNum" placeholder="Enter a number"/>
          <input id="btnFindPrimes" className="nambk-btn nambk-btn-primary" 
                 type="submit" value="Go"/>
        </div>
        <div id="result" className="mt-3" style={{minHeight: '4rem',}}>
          {this.state.answer.join(', ')}
        </div>   
      </form>
    );
  }
}
export default PrimesForm;