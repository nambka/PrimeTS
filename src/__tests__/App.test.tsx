import React from 'react';
import App from '../components/App';
import Header from '../components/Header';
import Footer from '../components/Footer';
import PrimeForm from '../components/PrimeForm';
import PrimesForm from '../components/PrimesForm';
import { shallow } from 'enzyme';

// https://feralamillo.medium.com/create-react-app-typescript-testing-with-jest-and-enzyme-869fdba1bd3
describe('App:', () => {
  it('renders components without errors', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Header)).toHaveLength(1);
    expect(wrapper.find('.container-fluid')).toHaveLength(1);
    expect(wrapper.find(PrimeForm)).toHaveLength(1);
    expect(wrapper.find(PrimesForm)).toHaveLength(1);
    expect(wrapper.find(Footer)).toHaveLength(1); 
    expect(wrapper).toMatchSnapshot();
    // console.log(wrapper.debug());
  });

  it('changes props when executing by calling them as functions with passed param 1', () => {
    const wrapper = shallow(<App />); 
    wrapper.find('Footer').prop('toggleTheme')({
      target: {
        theme: 'light'
      }
    });
    expect(wrapper.find('Footer').prop('theme')).toEqual('dark');
  });

  it('changes props when executing by calling them as functions with passed param 2', () => {
    const wrapper = shallow(<App />); 
    wrapper.find('Footer').prop('toggleTheme')({
      target: {
        theme: 'dark'
      }
    });
    expect(wrapper.find('Footer').prop('theme')).toEqual('moon');
  });

  it('changes props when executing by calling them as functions with passed param 3', () => {
    const wrapper = shallow(<App />); 
    wrapper.find('Footer').prop('toggleTheme')({
      target: {
        theme: 'moon'
      }
    });
    expect(wrapper.find('Footer').prop('theme')).toEqual('light');
  });
});
