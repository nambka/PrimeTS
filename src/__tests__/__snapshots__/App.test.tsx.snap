// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: renders components without errors 1`] = `
<Component
  theme={
    Object {
      "bodyBkg": "#F7F8F9",
      "bodyTextColor": "#353637",
      "footerBkg": "linear-gradient(150deg,#19a0ff,#2d1582)",
      "footerTextColor": "#fff",
      "formBkg": "#fff!important",
      "formBkgHover": "#fff",
      "formBorder": "1px solid #eaecef22",
      "formBoxShadow": "0 5px 30px 0 rgba(0,0,0,.2)",
      "formBoxShadowHover": "0 3px 10px 0 rgba(204,204,204,.8)",
      "formHeaderDescColor": "#006647",
      "formTransition": ".3s all ease",
      "formTransitionHover": ".2s all ease",
      "inputBkg": "#fff3",
      "inputBorder": "1px solid #E6E8EA",
      "inputBorderHover": "#F0B90B",
      "resultBoxShadow": "inset 0 0 5px 0 rgba(0,0,0,.1)",
    }
  }
>
  <Memo(l) />
  <Header />
  <main
    className="container-fluid text-center"
  >
    <PrimeForm />
    <PrimesForm />
  </main>
  <Footer
    theme="light"
    toggleTheme={[Function]}
  />
</Component>
`;
